### ISSUE TITLE
Produces an out-of-range result for MNIST

### ISSUE THREAD

**Issue by cliffbdf** [blackecho/Deep-Learning-TensorFlow#60]
Produces an out-of-range result (reconstruction loss) for MNIST.

Here is the console output:

[vagrant@machinelearning DeepAutoencoder-TensorFlow]$ make run
python ../Reference/Deep-Learning-TensorFlow/cmd_line/boltzmann/run_deep_autoencoder.py \
	--dataset mnist \
	--cifar_dir ../Reference/Datasets/MNIST/mnist.pkl \
	--main_dir deep-autoencoder \
	--model_name deeper-is-better \
	--rbm_layers 1000,500,250,100,30 \
	--rbm_batch_size 128 \
	--finetune_batch_size 128 \
	--finetune_loss_func cross_entropy \
	--verbose 1
Extracting MNIST_data/train-images-idx3-ubyte.gz
Extracting MNIST_data/train-labels-idx1-ubyte.gz
Extracting MNIST_data/t10k-images-idx3-ubyte.gz
Extracting MNIST_data/t10k-labels-idx1-ubyte.gz
Training layer 1...
2017-11-09 02:43:59.295164: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.1 instructions, but these are available on your machine and could speed up CPU computations.
2017-11-09 02:43:59.317348: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use SSE4.2 instructions, but these are available on your machine and could speed up CPU computations.
2017-11-09 02:43:59.317387: W tensorflow/core/platform/cpu_feature_guard.cc:45] The TensorFlow library wasn't compiled to use AVX instructions, but these are available on your machine and could speed up CPU computations.
Tensorboard logs dir for this run is /home/vagrant/.yadlt/logs/run6
Reconstruction loss: nan: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [08:26<00:00, 50.62s/it]
Training layer 2...
Tensorboard logs dir for this run is /home/vagrant/.yadlt/logs/run7
Reconstruction loss: nan: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [06:20<00:00, 38.10s/it]
Training layer 3...
Tensorboard logs dir for this run is /home/vagrant/.yadlt/logs/run8
Reconstruction loss: nan: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [01:25<00:00,  8.56s/it]
Training layer 4...
Tensorboard logs dir for this run is /home/vagrant/.yadlt/logs/run9
Reconstruction loss: nan: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:25<00:00,  2.55s/it]
Training layer 5...
Tensorboard logs dir for this run is /home/vagrant/.yadlt/logs/run10
Reconstruction loss: nan: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [00:08<00:00,  1.14it/s]
Tensorboard logs dir for this run is /home/vagrant/.yadlt/logs/run11
Reconstruction loss: nan: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████| 10/10 [17:52<00:00, 107.29s/it]
Test set reconstruction loss: nan
[vagrant@machinelearning DeepAutoencoder-TensorFlow]$
